File: python/app-testing-android.py
import os

from appium import webdriver
from appium.webdriver.common.touch_action import TouchAction

desired_cap = { 
  'platformName': 'android',
  'device': 'Google Nexus 6',
  'realMobile': True,
  'app': os.environ['ANDROID_APP_URL'],
  'build': 'App Testing Android',
  'name': 'sample python',
  'browserstack.debug': True
}

driver = webdriver.Remote(
  command_executor='http://'+os.environ['BROWSERSTACK_USER']+':'+os.environ['BROWSERSTACK_ACCESS_KEY']+'@hub.browserstack.com:80/wd/hub',
  desired_capabilities=desired_cap)


driver.get_screenshot_as_file("test.png")
sample_label = driver.find_element_by_id("sampleLabel")
print(sample_label.text)

context = driver.contexts

num1 = driver.find_element_by_id("num1")
num2 = driver.find_element_by_id("num2")
num1.send_keys("12")
num2.send_keys("32")
add_btn = driver.find_element_by_id("addBtn")

touch = TouchAction(driver)
touch.tap(add_btn).perform()

driver.get_screenshot_as_file("test.png")
print(sample_label.text)

driver.quit()

File: node/app-testing-android.js
var wd = require('wd');

// Input capabilities
var capabilities = {
  'platformName': 'android',
  'device': 'Google Nexus 6',
  'realMobile': true,
  'app': process.env.ANDROID_APP_URL,
  'build': 'App Testing Android',
  'name': 'sample node js',
  'browserstack.debug': true,
  'browserstack.user' : process.env.BROWSERSTACK_USER,
  'browserstack.key' : process.env.BROWSERSTACK_ACCESS_KEY
}

var driver = wd.promiseChainRemote({
  host: 'hub-cloud.browserstack.com',
  port: 80
});

driver
  .init(capabilities)
  .takeScreenshot().then(function(){})
  .elementById("sampleLabel")
  .text().then(function(t){
    console.log(t);
  })
  .contexts()
  .elementById("num1").sendKeys("12")
  .elementById("num2").sendKeys("32")
  .elementById("addBtn")
  .tap()
  .takeScreenshot()
  .elementById("sampleLabel")
  .text().then(function(t){
    console.log(t);
  })
  .quit()
  .done();
